Live Share Build 2023 updates (#8615)
MicrosoftDocs/msteams-docs: msteams-platform/apps-in-teams-meetings/teams-live-share-tutorial.md

    ? LiveShareHost.create()
    : TestLiveShareHost.create();
  // Create client
  const client = new LiveShareClient(host);
  // Join container
  return await client.joinContainer(containerSchema, onContainerFirstCreated);
}
```

> When testing locally, `TestLiveShareHost` updates the browser URL to contain the ID of the test container that was created. Copying that link to other browser tabs causes the `LiveShareClient` to join the test container that was created. If the modification of the applications URL interferers with the operation of the application, the strategy used to store the test containers ID can be customized using the [setLocalTestContainerId](/javascript/api/@microsoft/live-share/iliveshareclientoptions) and [getLocalTestContainerId](/javascript/api/@microsoft/live-share/iliveshareclientoptions) options passed to `LiveShareClient`.

Reviewer: ```suggestion
When testing locally, `TestLiveShareHost` updates the browser URL to contain the ID of the test container that was created. Copying that link to other browser tabs causes the `LiveShareClient` to join the test container that was created. If the modification of the applications URL interferes with the operation of the application, the strategy used to store the test containers ID can be customized using the [setLocalTestContainerId](/javascript/api/@microsoft/live-share/iliveshareclientoptions) and [getLocalTestContainerId](/javascript/api/@microsoft/live-share/iliveshareclientoptions) options passed to `LiveShareClient`.
```

  updateDice(1);
}
```

## Connect meeting stage view to Live Share

### Modify LiveState

To begin using Live Share in the application, the first thing to change is what happens when the user selects the `rollButton`. Instead of updating the local state directly, the button updates the number stored as a `state` value in `diceState`. Whenever you call `.set()` with a new `state`, that value is distributed to all clients. Any changes to the `diceState` can cause a `stateChanged` event to be emitted, and an event handler can trigger an update of the view.

This pattern is common in Fluid and Live Share distributed data structures because it enables the view to behave the same way for both local and remote changes.

```js
rollButton.onclick = () =>
  diceState.set(Math.floor(Math.random() * 6) + 1);
```

### Rely on Fluid data

The next change that needs to be made is to change the `updateDice` function to retrieve the latest dice value from the `LiveState` each time `updateDice` is called.

```js
const updateDice = () => {
  const diceValue = diceState.state;
  dice.textContent = String.fromCodePoint(0x267f + diceValue);
};
```

### Handle remote changes

> The values returned from `diceState` are only a snapshot in time. To keep the data up to date as it changes an event handler must be registered with the `diceState` to call `updateDice` each time that the `stateChanged` event is sent.

Reviewer: ```suggestion
The values returned from `diceState` are only a snapshot in time. To keep the data up-to-date as it changes, an event handler must be registered with `diceState` to call `updateDice` each time that the `stateChanged` event is sent.
```

```js
diceState.on("stateChanged", updateDice);
```

### Initialize LiveState

> Before you can begin receiving changes Live Share in the application, you must first call `initialize()` on your `LiveState` object with an initial value. This initial value will not overwrite any existing state that was sent by other users.

Reviewer: ```suggestion
Before you can begin receiving the Live Share changes in the application, you must first call `initialize()` on your `LiveState` object with an initial value. This initial value doesn't overwrite any existing state that was sent by other users.
```

Reviewer: @ryanbliss,

Please verify for technical accuracy.

> Once you've initialized `LiveState`, the `stateChanged` event you registered earlier will start to be triggered whenever a change is made. However, to update our UI with the initial value, next we will call `updateDice()`.

Reviewer: ```suggestion
After you've initialized `LiveState`, the `stateChanged` event you registered earlier starts to trigger whenever a change is made. However, to update the UI within the initial value, call `updateDice()`.
```
